newIDE/app/src/MainFrame/index.js

```diff
@@ -2351,9 +2407,14 @@ const MainFrame = (props: Props) => {
     ]
   );
 
+  const canSaveProjectAs = !!currentProject && !checkedOutVersionStatus;
   const saveProjectAs = React.useCallback(
     () => {
-      if (!currentProject) return;
+      if (!canSaveProjectAs) {
+        // Prevent "save project as" when no current project or when the opened project
+        // is a previous version (cloud project only) of the current project.
```

Comment:
Ubernit: I would move this comment just above `canSaveProjectAs`.

- When I read "const saveProjectAs = .... if (!canSaveProjectAs) return;", this is "obvious" that we early return if we can't save as - that's what the boolean says. Know if I want to know WHY this is the case, I go to the definition of the boolean...
- ...and at the definition of the boolean, that's where I would like an explanation about this boolean. "Ok, in which case this boolean is true/false? I can read the code but probably a comment is good here".

So this comment will be easier to find, especially if I try to find it because I stumbled upon "canSaveProjectAs" somewhere else in the file.